AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/QueryPredicate+SQLite.swift

```diff
@@ -50,3 +50,13 @@ extension QueryOperator {
         }
     }
 }
+
+extension QueryPredicate {
+    var isAll: Bool {
+        if let allPredicate = self as? QueryPredicateConstant, allPredicate == .all {
+            return true
+        } else {
+            return false
+        }
```

Comment:
```suggestion
        guard let allPredicate = self as? QueryPredicateConstant else {
            return false
        } 
        return allPredicate == .all
```


Will the above code cover if the query passed in is a `QueryPredicateGroup` with an `or QueryPredicateConstant.all` ?

Reply:
the code won't cover if the `.all` predicate is within a group like OR or AND. This change is added purely for the use case of explicitly passing in just `.all`. This is in discussion with @fjnoyp to match the same logic as Android, and so that allows Flutter's integration with native to work easily, since both native platforms have the same behavior. Passing in `.all` is the same as not passing anything in. From Flutter's prespective and according to Android's code, this is a valid use case. `.all` is a special predicate and I don't think it makes sense to do something like "condition = rating >5 or .all" since that reads "rating is greater than 5 or apply no condition", so that behavior is untested/undefined. Ideally we have compile time check on valid predicates as it would be hard to validate all invalid usages of predicates